### PR TITLE
Upgrade build-tools and pre-commit from template

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,7 @@ endif
 
 ifndef NOCOLOR
   GREEN  := $(shell tput -Txterm setaf 2)
-  YELLOW := $(shell tput -Txterm setaf 3)
+  YELLOW := $(shell tput -Txterm seaf 3)
```
Comment (mariux) [inline Makefile:28]: does not make sense to me?

**Review by mariux — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (soerenmartius) [reply Makefile]: sorry, my bad, that was a coding while driving with the bus through a bumpy mountain fuckup - should have checked the changes more properly. Just resolved this

**Review by mariux — APPROVED** [review]
lgtm

**Review by Hanymin — APPROVED** [review]
Thanks for the build-tools upgrade and adding the TF_version  check.
Lgtm

**Pull request merged**